### ISSUE TITLE
onFacesDetected breaks app on Android (device-only)

### ISSUE THREAD

**Issue by kittisakjitmon** [react-native-community/react-native-camera#2584]
Bug Report
To Do First

[x] Did you try latest release?
[x] Did you try master?
[x] Did you look for existing matching issues?

RNCamera

Steps to reproduce

Install react-native-camera with face detection using only manual procedures
Import { RNCamera, FaceDetector }
Add component to page with onFacesDetected callback
Actual behaviour
The app completely crash when using onFacesDetected but when not using it working normally.
On 'IOS' can face detect

Environment
React Native version: 0.61.2
react-native-camera version : 3.8.0

Version: npm version

Found this error in Android Studio
```
E/AndroidRuntime: FATAL EXCEPTION: AsyncTask #2
    Process: com.vertobase.mobile, PID: 25024
    java.lang.RuntimeException: An error occurred while executing doInBackground()
        at android.os.AsyncTask$3.done(AsyncTask.java:365)
        at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:383)
        at java.util.concurrent.FutureTask.setException(FutureTask.java:252)
        at java.util.concurrent.FutureTask.run(FutureTask.java:271)
        at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:257)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
        at java.lang.Thread.run(Thread.java:784)
     Caused by: java.lang.NoClassDefFoundError: Failed resolution of: Lcom/google/android/gms/internal/zzbfm;
        at org.reactnative.facedetector.RNFaceDetector.<init>(RNFaceDetector.java:29)
        at org.reactnative.camera.RNCameraView.setupFaceDetector(RNCameraView.java:338)
        at org.reactnative.camera.RNCameraView.setShouldDetectFaces(RNCameraView.java:375)
        at org.reactnative.camera.CameraViewManager.setFaceDetecting(CameraViewManager.java:152)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.facebook.react.uimanager.ViewManagersPropertyCache$PropSetter.updateViewProp(ViewManagersPropertyCache.java:83)
        at com.facebook.react.uimanager.ViewManagerPropertyUpdater$FallbackViewManagerSetter.setProperty(ViewManagerPropertyUpdater.java:132)
        at com.facebook.react.uimanager.ViewManagerPropertyUpdater.updateProps(ViewManagerPropertyUpdater.java:51)
        at com.facebook.react.uimanager.ViewManager.updateProperties(ViewManager.java:37)
        at com.facebook.react.uimanager.NativeViewHierarchyManager.updateProperties(NativeViewHierarchyManager.java:136)
        at com.facebook.react.uimanager.UIViewOperationQueue$UpdatePropertiesOperation.execute(UIViewOperationQueue.java:95)
        at com.facebook.react.uimanager.UIViewOperationQueue$1.run(UIViewOperationQueue.java:917)
        at com.facebook.react.uimanager.UIViewOperationQueue.flushPendingBatches(UIViewOperationQueue.java:1025)
        at com.facebook.react.uimanager.UIViewOperationQueue.access$2600(UIViewOperationQueue.java:46)
        at com.facebook.react.uimanager.UIViewOperationQueue$DispatchUIFrameCallback.doFrameGuarded(UIViewOperationQueue.java:1085)
        at com.facebook.react.uimanager.GuardedFrameCallback.doFrame(GuardedFrameCallback.java:29)
        at com.facebook.react.modules.core.ReactChoreographer$ReactChoreographerDispatcher.doFrame(ReactChoreographer.java:166)
        at com.facebook.react.modules.core.ChoreographerCompat$FrameCallback$1.doFrame(ChoreographerCompat.java:84)
        at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1039)
        at android.view.Choreographer.doCallbacks(Choreographer.java:847)
        at android.view.Choreographer.doFrame(Choreographer.java:771)
        at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:1027)
        at android.os.Handler.handleCallback(Handler.java:809)
        at android.os.Handler.dispatchMessage(Handler.java:102)
        at android.os.Looper.loop(Looper.java:166)
        at android.app.ActivityThread.main(ActivityThread.java:7555)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:469)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:963)
E/AndroidRuntime: Caused by: java.lang.ClassNotFoundException: Didn't find class "com.google.android.gms.internal.zzbfm" on path: DexPathList[[zip file "/data/app/com.vertobase.mobile-2J6NpwnK2eC8fM6CT9PxBA==/base.apk", zip file "/data/app/com.vertobase.mobile-2J6NpwnK2eC8fM6CT9PxBA==/split_lib_dependencies_apk.apk", zip file "/data/app/com.vertobase.mobile-2J6NpwnK2eC8fM6CT9PxBA==/split_lib_resources_apk.apk", zip file "/data/app/com.vertobase.mobile-2J6NpwnK2eC8fM6CT9PxBA==/split_lib_slice_0_apk.apk", zip file "/data/app/com.vertobase.mobile-2J6NpwnK2eC8fM6CT9PxBA==/split_lib_slice_1_apk.apk", zip file "/data/app/com.vertobase.mobile-2J6NpwnK2eC8fM6CT9PxBA==/split_lib_slice_2_apk.apk", zip file "/data/app/com.vertobase.mobile-2J6NpwnK2eC8fM6CT9PxBA==/split_lib_slice_3_apk.apk", zip file "/data/app/com.vertobase.mobile-2J6NpwnK2eC8fM6CT9PxBA==/split_lib_slice_4_apk.apk", zip file "/data/app/com.vertobase.mobile-2J6NpwnK2eC8fM6CT9PxBA==/split_lib_slice_5_apk.apk", zip file "/data/app/com.vertobase.mobile-2J6NpwnK2eC8fM6CT9PxBA==/split_lib_slice_6_apk.apk", zip file "/data/app/com.vertobase.mobile-2J6NpwnK2eC8fM6CT9PxBA==/split_lib_slice_7_apk.apk", zip file "/data/app/com.vertobase.mobile-2J6NpwnK2eC8fM6CT9PxBA==/split_lib_slice_8_apk.apk", zip file "/data/app/com.vertobase.mobile-2J6NpwnK2eC8fM6CT9PxBA==/split_lib_slice_9_apk.apk"],nativeLibraryDirectories=[/data/app/com.vertobase.mobile-2J6NpwnK2eC8fM6CT9PxBA==/lib/arm64, /data/app/com.vertobase.mobile-2J6NpwnK2eC8fM6CT9PxBA==/base.apk!/lib/arm64-v8a, /data/app/com.vertobase.mobile-2J6NpwnK2eC8fM6CT9PxBA==/split_lib_dependencies_apk.apk!/lib/arm64-v8a, /data/app/com.vertobase.mobile-2J6NpwnK2eC8fM6CT9PxBA==/split_lib_resources_apk.apk!/lib/arm64-v8a, /data/app/com.vertobase.mobile-2J6NpwnK2eC8fM6CT9PxBA==/split_lib_slice_0_apk.apk!/lib/arm64-v8a, /data/app/com.vertobase.mobile-2J6NpwnK2eC8fM6CT9PxBA==/split_lib_slice_1_apk.apk!/lib/arm64-v8a, /data/app/com.vertobase.mobile-2J6NpwnK2eC8fM6CT9PxBA==/split_lib_slice_2_apk.apk!/lib/arm64-v8a, /data/app/com.vertobase.mobile-2J6NpwnK2eC8fM6CT9PxBA==/split_lib_slice_3_apk.apk!/lib/arm64-v8a, /data/app/com.vertobase.mobile-2J6NpwnK2eC8fM6CT9PxBA==/split_lib_slice_4_apk.apk!/lib/arm64-v8a, /data/app/com.vertobase.mobile-2J6NpwnK2eC8fM6CT9PxBA==/split_lib_slice_5_apk.apk!/lib/arm64-v8a, /data/app/com.vertobase.mobile-2J6NpwnK2eC8fM6CT9PxBA==/split_lib_slice_6_apk.apk!/lib/arm64-v8a, /data/app/com.vertobase.mobile-2J6NpwnK2eC8fM6CT9PxBA==/split_lib_slice_7_apk.apk!/lib/arm64-v8a, /data/app/com.vertobase.mobile-2J6NpwnK2eC8fM6CT9PxBA==/split_lib_slice_8_apk.apk!/lib/arm64-v8a, /data/app/com.vertobase.mobile-2J6NpwnK2eC8fM6CT9PxBA==/split_lib_slice_9_apk.apk!/lib/arm64-v8a, /system/lib64, /vendor/lib64, /product/lib64]]
        at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:125)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:379)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:312)
        	... 29 more
    	Suppressed: java.io.IOException: No original dex files found for dex location /data/app/com.vertobase.mobile-2J6NpwnK2eC8fM6CT9PxBA==/split_lib_resources_apk.apk
        at dalvik.system.DexFile.openDexFileNative(Native Method)
        at dalvik.system.DexFile.openDexFile(DexFile.java:353)
        at dalvik.system.DexFile.<init>(DexFile.java:100)
        at dalvik.system.DexFile.<init>(DexFile.java:74)
        at dalvik.system.DexPathList.loadDexFile(DexPathList.java:374)
        at dalvik.system.DexPathList.makeDexElements(DexPathList.java:337)
        at dalvik.system.DexPathList.<init>(DexPathList.java:157)
        at dalvik.system.BaseDexClassLoader.<init>(BaseDexClassLoader.java:65)
        at dalvik.system.PathClassLoader.<init>(PathClassLoader.java:64)
        at com.android.internal.os.ClassLoaderFactory.createClassLoader(ClassLoaderFactory.java:73)
        at com.android.internal.os.ClassLoaderFactory.createClassLoader(ClassLoaderFactory.java:88)
        at android.app.ApplicationLoaders.getClassLoader(ApplicationLoaders.java:75)
        at android.app.ApplicationLoaders.getClassLoader(ApplicationLoaders.java:37)
E/AndroidRuntime:     at android.app.LoadedApk.createOrUpdateClassLoaderLocked(LoadedApk.java:735)
        at android.app.LoadedApk.getClassLoader(LoadedApk.java:769)
        at android.app.LoadedApk.getResources(LoadedApk.java:1022)
        at android.app.ContextImpl.createAppContext(ContextImpl.java:2537)
        at android.app.ActivityThread.handleBindApplication(ActivityThread.java:6526)
        at android.app.ActivityThread.-wrap2(Unknown Source:0)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2084)
        at android.os.Handler.dispatchMessage(Handler.java:109)
        		... 5 more
```

**Comment by cristianoccazinsp** [comment]
`Caused by: java.lang.IllegalStateException: Default FirebaseApp is not initialized in this process com.rncamera. Make sure to call FirebaseApp.initializeApp(Context) first.`

That should tell you something. I haven't used this at all so I can't really help, but looks like a configuration issue? Did you follow all the steps correctly?

**Comment by KittisakChitmun** [comment]
1. `npm install react-native-camera --save`
2. Open up `android/app/src/main/java/[...]/MainApplication.java`
 - Add `import org.reactnative.camera.RNCameraPackage;` to the imports at the top of the file
 - Add `new RNCameraPackage()` to the list returned by the `getPackages()` method. Add a comma to the previous item if there's already something there.
3. Append the following lines to `android/settings.gradle:`
```
include ':react-native-camera'
project(':react-native-camera').projectDir = new File(rootProject.projectDir, 	'../node_modules/react-native-camera/android')
```

4.Insert the following lines in `android/app/build.gradle:`
```gradle
implementation project(':react-native-camera')
```
 if using MLKit for text/face/barcode recognition:
```
android {
  ...
  defaultConfig {
    ...
    missingDimensionStrategy 'react-native-camera', 'mlkit' <-- insert this line
  }
}
```
5.Declare the permissions in your Android Manifest (required for video recording feature)
```
<uses-permission android:name="android.permission.RECORD_AUDIO"/>
<uses-permission android:name="android.permission.CAMERA" />
<uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE" />
<uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE" />
```
6.Add jitpack to android/build.gradle
```
allprojects {
    repositories {
        maven { url "https://www.jitpack.io" }
        maven { url "https://maven.google.com" }
    }
}
```
7. Additional steps for using MLKit for text/face/barcode recognition

   7.1. Using Firebase MLKit requires seting up Firebase project for your app. If you have not already 
   added Firebase to your app, please follow the steps described in getting started guide. In short, 
   you would need to

- Register your app in Firebase console.
- Download google-services.json and place it in `android/app/`
- add the folowing to project level build.gradle:
```
buildscript {
      dependencies {
      // Add this line
      classpath 'com.google.gms:google-services:4.0.1' <-- you might want to use different version
      }
    }
```
add to the bottom of `android/app/build.gradle` file
```
apply plugin: 'com.google.gms.google-services'
```
   7.2. Configure your app to automatically download the ML model to the device after your app is 
   installed from the Play Store. If you do not enable install-time model downloads, the model will be 
   downloaded the first time you run the on-device detector. Requests you make before the 
   download has completed will produce no results.
```
<application ...>
...
  <meta-data
      android:name="com.google.firebase.ml.vision.DEPENDENCIES"
      android:value="ocr, face, barcode" />
</application>
```

**Comment by KarlHeitmann** [comment]
How did you write the callback in your app? I've also had the same issue, I wrote the callback on onFacesDetected and the app crashed. This was the wrong code:


`
class OcularCam extends PureComponent {
  handleFaceDetected = (argumentos) => {
    console.log(argumentos)
  }
  ........
  render() {
    return(
    ....
        <RNCamera
          .......
          onFacesDetected={this.handleFaceDetected}
        />
    ....
    )
  }
}
`

I've decided to put an arrow function directly into the brackets of the props onFacesDetected, like this:
`
<RNCamera
  ....
  onFacesDetected={( argumento ) => {
    console.log(argumento)
    console.log(argumento.faces[0].bounds)
  }
/>
`

This way, the problem was solved.
